src/main/java/athena/TaskList.java

```diff
@@ -5,14 +5,23 @@
 public class TaskList {
     public static final String NO_FILTER = "";
     private ArrayList<Task> tasks;
+    private int maxIndex = -1;
 
     public TaskList() {
         tasks = new ArrayList<>();
     }
 
     public TaskList(ArrayList<Task> taskList) {
         tasks = new ArrayList<>();
-        for (Task task: taskList) {
+        for (Task task : taskList) {
+            tasks.add(task);
+        }
+    }
+
+    public TaskList(ArrayList<Task> taskList, int maxIndex) {
+        this.maxIndex = maxIndex;
+        tasks = new ArrayList<>();
+        for (Task task : taskList) {
             tasks.add(task);
```

Comment:
Code here is the same as lines 15-17. We can do `this(taskList)` to avoid repetition.

Reply:
I refactored this part and deleted the overloading because it was not being used